internal/security/bootstrapper/command/setupacl/tokencleaner/tokencleaner.go

```diff
@@ -0,0 +1,253 @@
+
+		return nil
+	})
+
+	return walkErr
+}
+
+func (tr *TokenCleaner) readTokenFromFile(filePath string) (string, error) {
+	fileOpener := fileioperformer.NewDefaultFileIoPerformer()
+	reader, err := fileOpener.OpenFileReader(filePath, os.O_RDONLY, 0400)
+	if err != nil {
+		return share.EmptyToken, fmt.Errorf("failed to open file reader: %v", err)
+	}
+
+	token, err := ioutil.ReadAll(reader)
+	if err != nil {
+		return share.EmptyToken, fmt.Errorf("failed to read token from reader: %v", err)
+	}
+
+	return strings.TrimSpace(string(token)), nil
+}
+
+func (tr *TokenCleaner) retrieveSelfToken(filepath, token string) (*SelfTokenInfo, error) {
+	if len(token) == 0 {
+		tr.lc.Infof("tokenID is empty for path %s, skip readTokenSelf", filepath)
+		return nil, nil
+	}
+
+	readTokenSelfURL := tr.tokenApiUrlBase + consulReadSelfTokenAPI
+	req, err := http.NewRequest(http.MethodGet, readTokenSelfURL, http.NoBody)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to prepare readTokenSelfURL request for http URL: %w", err)
+	}
+
+	req.Header.Add(share.ConsulTokenHeader, token)
+	resp, err := tr.httpCaller.Do(req)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to send readTokenSelfURL request for http URL: %w", err)
+	}
+
+	defer func() {
+		_ = resp.Body.Close()
+	}()
+
+	tokenSelfResp, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to read readTokenSelfURL response body: %w", err)
+	}
+
+	switch resp.StatusCode {
+	case http.StatusOK:
+		var tokenSelf SelfTokenInfo
+		if err := json.NewDecoder(bytes.NewReader(tokenSelfResp)).Decode(&tokenSelf); err != nil {
+			return nil, fmt.Errorf("failed to decode ConsulToken json data: %v", err)
+		}
+
+		return &tokenSelf, nil
+	case http.StatusForbidden:
+		// in this case, the self token is not found, so we just return nil
+		return nil, nil
+	default:
+		return nil, fmt.Errorf("failed to read self token for path %s with status code= %d: %s",
+			filepath, resp.StatusCode, string(tokenSelfResp))
+	}
+}
+
+func (tr *TokenCleaner) deleteToken(path, tokenAccessorID, bootstrapACLToken string) error {
+	if len(tokenAccessorID) == 0 {
+		tr.lc.Infof("tokenAccessorID is empty for path %s, skip deleteToken", path)
+		return nil
+	}
+
+	delateTokenURL := tr.tokenApiUrlBase + fmt.Sprintf(consulDeleteTokenAPI, tokenAccessorID)
+	req, err := http.NewRequest(http.MethodDelete, delateTokenURL, http.NoBody)
+	if err != nil {
+		return fmt.Errorf("Failed to prepare delateTokenURL request for http URL: %w", err)
+	}
+
+	req.Header.Add(share.ConsulTokenHeader, bootstrapACLToken)
+	resp, err := tr.httpCaller.Do(req)
+	if err != nil {
+		return fmt.Errorf("Failed to send delateTokenURL request for http URL: %w", err)
+	}
+
+	defer func() {
+		_ = resp.Body.Close()
+	}()
+
+	tokenSelfResp, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return fmt.Errorf("Failed to read delateTokenURL response body: %w", err)
+	}
+
+	switch resp.StatusCode {
```

Comment:
What code is returned if you try to delete an already-deleted or non-existent token?

Reply:
Two different cases:
depending on timing, i have observed that if deleting in repetitive manner, for the deleted one will be sometimes 200 
- delete an already-deleted token: 200, and "true"; it seems to me Consul has some cache:
look at the following, I deleted the same accesorID two in a row (i.e., issued two same delete request twice)
```
jim@jim-NUC7i5DNHE:~/go/src/github.com/edgexfoundry/edgex-compose/compose-builder$ curl -v -H "X-Consul-Token:b34b9cb9-f3a5-99dc-f662-90b060247b64" -X DELETE http://localhost:8500/v1/acl/token/c3887884-20ee-0368-1540-1def3565bcc7
*   Trying 127.0.0.1...
* Connected to localhost (127.0.0.1) port 8500 (#0)
> DELETE /v1/acl/token/c3887884-20ee-0368-1540-1def3565bcc7 HTTP/1.1
> Host: localhost:8500
> User-Agent: curl/7.47.0
> Accept: */*
> X-Consul-Token:b34b9cb9-f3a5-99dc-f662-90b060247b64
> 
< HTTP/1.1 200 OK
< Content-Type: application/json
< Vary: Accept-Encoding
< X-Consul-Default-Acl-Policy: allow
< Date: Tue, 30 Mar 2021 22:28:17 GMT
< Content-Length: 4
< 
* Connection #0 to host localhost left intact
truejim@jim-NUC7i5DNHE:~/go/src/github.com/edgexfoundry/edgex-compose/compose-builder$ curl -v -H "X-Consul-Token:b34b9cb9-f3a5-99dc-f662-90b060247b64" -X DELETE http://localhost:8500/v1/acl/token/c388720ee-0368-1540-1def3565bcc7
*   Trying 127.0.0.1...
* Connected to localhost (127.0.0.1) port 8500 (#0)
> DELETE /v1/acl/token/c3887884-20ee-0368-1540-1def3565bcc7 HTTP/1.1
> Host: localhost:8500
> User-Agent: curl/7.47.0
> Accept: */*
> X-Consul-Token:b34b9cb9-f3a5-99dc-f662-90b060247b64
> 
< HTTP/1.1 200 OK
< Content-Type: application/json
< Vary: Accept-Encoding
< X-Consul-Default-Acl-Policy: allow
< Date: Tue, 30 Mar 2021 22:28:20 GMT
< Content-Length: 4
< 
* Connection #0 to host localhost left intact
 curl -v -H "X-Consul-Token:b34b9cb9-f3a5-99dc-f662-90b060247b64" -X DELETE http://localhost:8500/v1/acl/token/c388720ee-0368-1540-1def3565bcc7
*   Trying 127.0.0.1...
* Connected to localhost (127.0.0.1) port 8500 (#0)
> DELETE /v1/acl/token/c388720ee-0368-1540-1def3565bcc7 HTTP/1.1
> Host: localhost:8500
> User-Agent: curl/7.47.0
> Accept: */*
> X-Consul-Token:b34b9cb9-f3a5-99dc-f662-90b060247b64
> 
< HTTP/1.1 500 Internal Server Error
< Vary: Accept-Encoding
< X-Consul-Default-Acl-Policy: allow
< Date: Tue, 30 Mar 2021 22:38:30 GMT
< Content-Length: 41
< Content-Type: text/plain; charset=utf-8
< 
* Connection #0 to host localhost left intact
Accessor ID is missing or an invalid UUID
```

- Delete a non-existent token:  it returns 500 if your accessor ID is gone OR your accesorID is not even exists.

```
curl -v -H "X-Consul-Token:b34b9cb9-f3a5-99dc-f662-90b060247b64" -X DELETE http://localhost:8500/v1/acl/token/13884-20ee-0368-1540-777777777777
*   Trying 127.0.0.1...
* Connected to localhost (127.0.0.1) port 8500 (#0)
> DELETE /v1/acl/token/c38813884-20ee-0368-1540-777777777777 HTTP/1.1
> Host: localhost:8500
> User-Agent: curl/7.47.0
> Accept: */*
> X-Consul-Token:b34b9cb9-f3a5-99dc-f662-90b060247b64
> 
< HTTP/1.1 500 Internal Server Error
< Vary: Accept-Encoding
< X-Consul-Default-Acl-Policy: allow
< Date: Tue, 30 Mar 2021 22:34:54 GMT
< Content-Length: 41
< Content-Type: text/plain; charset=utf-8
< 
* Connection #0 to host localhost left intact
Accessor ID is missing or an invalid UUID
```